compiler/qsc/src/interpret/tests.rs

```diff
@@ -2089,5 +2089,30 @@ mod given_interpreter {
                 entry: None,
             }
         }
+
+        #[test]
+        fn name_resolution_from_source_named_main_should_succeed() {
+            let sources = SourceMap::new(
+                [(
+                    "Main".into(),
+                    r#"function Foo() : Unit { Message("hello there..."); }"#.into(),
+                )],
+                None,
+            );
+            let (std_id, store) =
+                crate::compile::package_store_with_stdlib(TargetCapabilityFlags::all());
+            let mut interpreter = Interpreter::new(
+                sources,
+                PackageType::Lib,
+                TargetCapabilityFlags::all(),
+                LanguageFeatures::default(),
+                store,
+                &[(std_id, None)],
+            )
+            .expect("interpreter should be created");
+
+            let (result, output) = line(&mut interpreter, "Foo()");
```

Comment:
I was expecting `Main.Foo()` to be the correctly qualified name here, and I wouldn't expect `Foo()` to resolve at all. After all that's how it works when you reference `Main` items from a different Q# file.

Reply:
We dug into this, and you are right that this case should be an error. With the much nicer fix we landed at, I've pulled this out into a separate negative test.